src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -157,6 +186,18 @@ public boolean removePerson(ReadOnlyPerson key) throws PersonNotFoundException {
         }
     }
 
+    /**
+     * Removes {@code key} from this {@code AddressBook}.
+     * @throws PersonNotFoundException if the {@code key} is not in this {@code AddressBook}.
+     */
+    public boolean removeBlacklistedPerson(ReadOnlyPerson key) throws PersonNotFoundException {
+        if (blacklistedPersons.remove(key)) {
+            return true;
+        } else {
+            throw new PersonNotFoundException();
```

Comment:
Perhaps this piece of code can be further refined without needing the else statement

```java
if (blacklistedPersons.remove(key)) {
            return true;
}
throw new PersonNotFoundException();
```